app/src/main/kotlin/com/mapbox/vision/teaser/MainActivity.kt

```diff
@@ -566,13 +598,33 @@ class MainActivity : AppCompatActivity(), ReplayModeFragment.OnSelectModeItemLis
 
         VisionSafetyManager.create(VisionReplayManager)
         VisionSafetyManager.visionSafetyListener = visionSafetyListener
+
+        playback_seek_bar_view.setDuration(VisionReplayManager.getDuration())
+        playback_seek_bar_view.onSeekBarChangeListener = object : SeekBar.OnSeekBarChangeListener {
+
+            override fun onProgressChanged(seekBar: SeekBar?, progress: Int, fromUser: Boolean) {
+                if (fromUser) {
+                    VisionReplayManager.setProgress(progress.toFloat())
+                }
+            }
+
+            override fun onStartTrackingTouch(seekBar: SeekBar?) {
+                playbackTracked = true
+            }
+
+            override fun onStopTrackingTouch(seekBar: SeekBar?) {
+                playbackTracked = false
+            }
+        }
+
         return true
     }
 
     private fun destroyVisionManagerReplay() {
         VisionSafetyManager.destroy()
         VisionReplayManager.stop()
         VisionReplayManager.destroy()
+        playback_seek_bar_view.onSeekBarChangeListener = null
```

Comment:
I'd put it outside of manager deinit fun

Reply:
moved to parent function